R/melt_table.R

```diff
@@ -25,16 +25,9 @@
 #' @inheritParams read_table
 #' @export
 #' @examples
-#' # One corner from http://www.masseyratings.com/cf/compare.htm
-#' massey <- readr_example("massey-rating.txt")
-#' cat(read_file(massey))
-#' melt_table(massey)
-#'
-#' # Sample of 1978 fuel economy data from
-#' # http://www.fueleconomy.gov/feg/epadata/78data.zip
-#' epa <- readr_example("epa78.txt")
-#' cat(read_file(epa))
-#' melt_table(epa)
+#' ws <- readr_example("whitespace-sample.txt")
+#' writeLines(read_file(ws))
+#' melt_table(ws)
```

Comment:
```suggestion
#' fwf <- readr_example("fwf-sample.txt")
#' writeLines(read_lines(fwf))
#' melt_table(fwf)
#' 
#' ws <- readr_example("whitespace-sample.txt")
#' writeLines(read_lines(ws))
#' melt_table2(ws)
```

Reply:
Remember the overall goal is to demonstrate readr functions with an example dataset that shows the most suitable input for the function. The discovery of #1333 is that these examples were originally written when readr behaved differently. Then the readr functions changed, but the examples didn't, which left behind some rather nonsensical example code.

The `melt*()` functions are designed to return output where "each row represents a single token".

I think `melt_table()` is best paired with `fwf-sample.txt`. `melt_table2()` makes much more sense for `whitespace-sample.txt`. 

Also, from playing around with this, I think `read_lines()` is generally what we want (not `read_file()`).

``` r
devtools::load_all()
#> ℹ Loading readr

fwf <- readr_example("fwf-sample.txt")
writeLines(read_lines(fwf))
#> John Smith          WA        418-Y11-4111
#> Mary Hartford       CA        319-Z19-4341
#> Evan Nolan          IL        219-532-c301
melt_table(fwf)
#> Warning: `melt_table()` was deprecated in readr 2.0.0.
#> Please use `meltr::melt_table()` instead
#> # A tibble: 12 × 4
#>      row   col data_type value       
#>    <dbl> <dbl> <chr>     <chr>       
#>  1     1     1 character John        
#>  2     1     2 character Smith       
#>  3     1     3 character WA          
#>  4     1     4 character 418-Y11-4111
#>  5     2     1 character Mary        
#>  6     2     2 character Hartford    
#>  7     2     3 character CA          
#>  8     2     4 character 319-Z19-4341
#>  9     3     1 character Evan        
#> 10     3     2 character Nolan       
#> 11     3     3 character IL          
#> 12     3     4 character 219-532-c301

ws <- readr_example("whitespace-sample.txt")
writeLines(read_lines(ws))
#> first last state phone
#> John Smith WA 418-Y11-4111
#> Mary Hartford CA 319-Z19-4341
#> Evan Nolan IL 219-532-c301

# this doesn't make much sense as an example
# it does not produce one token per row
melt_table(ws)
#> Warning: `melt_table()` was deprecated in readr 2.0.0.
#> Please use `meltr::melt_table()` instead
#> # A tibble: 4 × 4
#>     row   col data_type value                        
#>   <dbl> <dbl> <chr>     <chr>                        
#> 1     1     1 character first last state phone       
#> 2     2     1 character John Smith WA 418-Y11-4111   
#> 3     3     1 character Mary Hartford CA 319-Z19-4341
#> 4     4     1 character Evan Nolan IL 219-532-c301

# this does
melt_table2(ws)
#> Warning: `melt_table2()` was deprecated in readr 2.0.0.
#> Please use `meltr::melt_table2()` instead
#> # A tibble: 16 × 4
#>      row   col data_type value       
#>    <dbl> <dbl> <chr>     <chr>       
#>  1     1     1 character first       
#>  2     1     2 character last        
#>  3     1     3 character state       
#>  4     1     4 character phone       
#>  5     2     1 character John        
#>  6     2     2 character Smith       
#>  7     2     3 character WA          
#>  8     2     4 character 418-Y11-4111
#>  9     3     1 character Mary        
#> 10     3     2 character Hartford    
#> 11     3     3 character CA          
#> 12     3     4 character 319-Z19-4341
#> 13     4     1 character Evan        
#> 14     4     2 character Nolan       
#> 15     4     3 character IL          
#> 16     4     4 character 219-532-c301
```

<sup>Created on 2022-01-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>